.github/workflows/maven.yml

```diff
@@ -0,0 +1,74 @@
+name: Arquillian-Cube Maven Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - 'doc/**'
+      - '.circleci/**'
+env:
+  # failsafe.groups configuration depends on the env setup.
+  FAILSAFE_GROUPS: ''
+  DOCKER_HOST: 'unix:///var/run/docker.sock'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ['11']
+
+    steps:
+      - name: Setup Minikube
+        uses: manusa/actions-setup-minikube@v2.10.0
+        with:
+          driver: docker
+          container runtime: containerd
+          minikube version: 'v1.32.0'
+          kubernetes version: 'v1.30.0'
+          github token: ${{ secrets.GITHUB_TOKEN }}
+          start args: "--memory='4gb' --cpus='2'"
+        env:
+          FAILSAFE_GROUPS: "-Dfailsafe.groups=\"org.arquillian.cube.docker.impl.requirement.RequiresDocker,org.arquillian.cube.kubernetes.impl.requirement.RequiresKubernetes\""
+      - name: Enable minikube registry
+        run: |
+          minikube addons enable registry
+          kubectl port-forward --namespace kube-system service/registry 5000:80 &
+      - name: Checkout arquillian-cube
+        uses: actions/checkout@v4
+      - uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'adopt'
+      - name: Maven pre-fetch dependencies ${{ matrix.java }}
+        run: |
+          ./mvnw verify -q -U -DskipTests # pre-fetch dependencies
+      - name: Build with Maven Java ${{ matrix.java }}
+        run: |
+          ./mvnw -fae clean package ${{ env.FAILSAFE_GROUPS }}
+      - uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: surefire-reports-${{ matrix.os }}-${{ matrix.java }}
+          path: '${{ github.workspace }}/**/surefire-reports/*.*'
+      - uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: server-logs-${{ matrix.os }}-${{ matrix.java }}
```

Comment:
The ${{ matrix.os }} would fail to resolve now.

Reply:
right !